hphp/runtime/base/mixed-array-defs.h

```diff
@@ -113,6 +125,20 @@ inline void MixedArray::initHash(int32_t* hash, uint32_t scale) {
     "ja         .l%=\n"
     : "+r"(offset) : "r"(hash) : "xmm0"
   );
+#elif defined(__aarch64__)
+  static_assert(Empty == -1, "The following fills with all 1's.");
+  assertx(HashSize(scale) == scale * 4);
+
+  uint64_t offset = scale * 16;
+  uint64_t ones = -1;
+  auto hash2 = hash;
+  __asm__ __volatile__(
+    ".l%=:\n"
+    "subs       %x0, %x0, #16\n"
+    "stp        %x2, %x2, [%x1], #16\n"
+    "bhi        .l%=\n"
```

Comment:
It might be better if you swap around the subs and stp.  As on Cavium's cores, the subs can be issued directly with the branch and not take up a cycle hit.